scio-core/src/main/scala/com/spotify/scio/schemas/instances/JavaInstances.scala

```diff
@@ -69,4 +70,21 @@ trait JavaInstances {
   implicit def javaBeanSchema[T: IsJavaBean: ClassTag]: RawRecord[T] =
     RawRecord[T](new JavaBeanSchema())
 
+  implicit def javaEnumSchema[T <: java.lang.Enum[T] : ClassTag]: Schema[T] =
+    Type[T](FieldType.logicalType(new LogicalType[T, String] {
+      private val clazz = scala.reflect.classTag[T].runtimeClass
+      private val className = clazz.getCanonicalName
+      private val values: Map[String, T] =
+        clazz.getMethod("values")
+          .invoke(null).asInstanceOf[Array[T]]
+          .map{ e => (e.name(), e) }
+          .toMap
+      override def getIdentifier: String = className
+      override def getBaseType: FieldType = FieldType.STRING
+      override def toBaseType(input: T): String = input.name()
+      override def toInputType(base: String): T = values(base)
```

Comment:
awesome! I think we can even reduce this by just using:
 
```suggestion
      override def toInputType(base: String): T = java.lang.Enum.valueOf(clazz, base)
```

Reply:
Good catch. Apparently I can't Java 😅